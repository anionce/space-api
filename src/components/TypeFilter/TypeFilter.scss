@import '../../assets/styles/variables';

.select-container {
	margin-top: 24px;
	display: flex;
	flex-direction: column;
	align-items: center;
}
