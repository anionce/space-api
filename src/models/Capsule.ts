export type Capsule = {
	id: string;
	land_landings: number;
	last_update: string;
	launches: string[];
	reuse_count: number;
	serial: string;
	status: string;
	type: string;
	water_landings: number;
};

export enum CapsuleType {
	Dragon10 = 'Dragon 1.0',
	Dragon11 = 'Dragon 1.1',
	Dragon20 = 'Dragon 2.0',
}

export const capsuleFilterTypes: CapsuleType[] = [CapsuleType.Dragon10, CapsuleType.Dragon11, CapsuleType.Dragon20];
