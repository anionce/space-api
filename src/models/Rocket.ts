export type Rocket = {
	id: string | undefined;
	serial: string;
	type: string | undefined;
};

export enum RocketType {
	SuperRocket = 'SuperRocket',
	ExtraRocket = 'ExtraRocket',
	HiperRocket = 'HiperRocket',
}

export const rocketFilterTypes: RocketType[] = [RocketType.SuperRocket, RocketType.HiperRocket, RocketType.ExtraRocket];
