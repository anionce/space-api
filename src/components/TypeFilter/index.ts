import { TypeFilter } from './TypeFilter';
export { TypeFilter };
