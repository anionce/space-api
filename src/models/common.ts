import { Capsule, CapsuleType } from './Capsule';
import { Rocket, RocketType } from './Rocket';

export type FilterTypes = RocketType | CapsuleType;

export type Item = Rocket | Capsule;
