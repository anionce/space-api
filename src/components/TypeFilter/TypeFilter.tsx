import React from 'react';
import { ChangeEvent } from 'react';
import { useTranslation } from 'react-i18next';
import { FilterTypes } from '../../models/common';
import './TypeFilter.scss';

type typeFilterProps = {
	onTypeChanged: (event: ChangeEvent<HTMLSelectElement>) => void;
	itemFilterTypes: FilterTypes[];
	typeSelected?: string;
	defaultOption?: string;
};

export const TypeFilter = ({ onTypeChanged, typeSelected, itemFilterTypes, defaultOption }: typeFilterProps) => {
	const { t } = useTranslation();

	return (
		<div className='select-container'>
			<label htmlFor='type'>{t('space_api.selector.type')}</label>
			<select id='type' onChange={onTypeChanged} defaultValue={typeSelected ?? ''}>
				<option value={defaultOption}>{defaultOption}</option>
				{itemFilterTypes.map((option: FilterTypes) => (
					<option key={option} value={option}>
						{option}
					</option>
				))}
			</select>
		</div>
	);
};
