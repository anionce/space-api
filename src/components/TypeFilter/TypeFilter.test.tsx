import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { screen } from '@testing-library/react';
import { customRender } from '../../assets/tests/utils';
import { TypeFilter } from './TypeFilter';
import { rocketFilterTypes } from '../../models/Rocket';
import { RocketType } from 'src/models/Rocket';

const defaultProps = {
	onTypeChanged: jest.fn(),
	itemFilterTypes: rocketFilterTypes,
};

test('renders correcly', () => {
	const { container } = customRender(<TypeFilter {...defaultProps} />);

	expect(container).toMatchSnapshot();
});

test('all elements are displayed', () => {
	customRender(<TypeFilter {...defaultProps} />);

	expect(screen.getByText(RocketType.SuperRocket)).toBeInTheDocument();
	expect(screen.getByText(RocketType.ExtraRocket)).toBeInTheDocument();
	expect(screen.getByText(RocketType.HiperRocket)).toBeInTheDocument();
});
