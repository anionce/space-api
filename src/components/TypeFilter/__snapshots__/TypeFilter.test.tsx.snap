// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correcly 1`] = `
<div>
  <div
    class="select-container"
  >
    <label
      for="type"
    >
      Type
    </label>
    <select
      id="type"
    >
      <option
        selected=""
      />
      <option
        value="SuperRocket"
      >
        SuperRocket
      </option>
      <option
        value="HiperRocket"
      >
        HiperRocket
      </option>
      <option
        value="ExtraRocket"
      >
        ExtraRocket
      </option>
    </select>
  </div>
</div>
`;
